### DISCUSSION TITLE
Configure Zitadel for Synology DSM

### DISCUSSION THREAD

**Discussion by Aurel004** [zitadel/zitadel#7907] [Show and tell]
This tutorial will work for Synology users that run DSM v7.2+ as you can use local existing users in DSM rather than use a LDAP Server to connect.

1. Go to your zitadel console: https://zitadel.mydomain.com/ui/console

2. Go to Projects and create a new one if needed, mine will be `server`

3. Click on New

4. Set a name, like _DSM_ and choose `WEB`
![image](https://github.com/zitadel/zitadel/assets/20973812/71f694eb-f922-428e-8b6e-81ac50c383a7)

5. Select `Code`
![image](https://github.com/zitadel/zitadel/assets/20973812/8fd628dd-fdb0-4000-b8fa-20893a776f59)

6. In _Redirect URIs_ put your DSM domain like `https://dsm.mydomain.com`
![image](https://github.com/zitadel/zitadel/assets/20973812/2ccf5f7f-86a3-4a01-b403-0e4f81e0374e)

7. Click _Create_ and copy the _ClientId_ and _ClientSecret_ provided to Notepad

8. You will need to provide the username of your Zitadel account to DSM
    - Click on your newly created app _DSM_
    - Click _Token Settings_ and tick _User Info inside ID Token_
![image](https://github.com/zitadel/zitadel/assets/20973812/e1b1771c-7496-47f6-b79b-796fa1282944)

9. Go to DSM, login, open _Control Panel_, click _Domain/LDAP_ and click on _SSO Client_

10. Tick `Select SSO by default on the login page` page if you want, then click on _Enable OpenID Connect SSO service_

11. Click the _OpenID Connect SSO Settings_ button

12. Configuration
    - Profile: `OIDC`
    - Account type: `Domain/LDAP/local`
    - Name: `Zitadel` or any name you want on the login page, it can be `OAuth`
    - Well-known URL: `https://zitadel.mydomain.com/.well-known/openid-configuration`
    - Application ID: `XXXX@server` the _ClientId_ copied in step 7
    - Application Secret: `XXXX` the _ClientSecret_  copied in step 7
    - Redirect URI: `https://dsm.mydomain.com` must match the redirect URI set in step 6
    - Authorisation scope: `openid profile preferred_username`
    - Username claim: `preferred_username`

### What if your username in Zitadel is not the same username in DSM ?
The _preferred_username_ as set above, will use your username set in Zitadel, you can see it on your profile page in Zitadel.
If your username is not the same and you do not want to change it, you can tweak it to set a custom username.
As Zitadel does not provide (yet) custom scopes, you can set the _Nickname_ field in Zitadel to match your DSM account username

1. Go back to Zitadel and click on your profile to _Edit Account_
2. In _Nickname_ set your DSM username like `myDSMUsername`
3. Click _Save_
4. In the OIDC settings in DSM, update:
    - Authorisation scope: `openid profile preferred_username` to  `openid profile nickname`
    - Username claim: `preferred_username` to `nickname`

And voilà, you're done!

**Comment by hifabienne** [comment]
Hei @Aurel004 
Thanks for putting this together and let the community know how it works. 🥳

**Comment by draeron** [comment]
anyone knows how to get the SSO working with multiple redirect URLs ?

```
024-11-16T13:54:54-05:00 ebola synoscgi_SYNO.API.Auth_7_login[2772]: sso.cpp:56 request token uri curl: The requested URL returned error: 400.
```

I set up a 2nd redirect to enable SSO login for the drive webapp but it seems Synology only use the first URL in it's config...

**Reply by fforootd** [reply]
Can you share how you configured both sides?

**Reply by draeron** [reply]
the issue seems to be on the synology side, the oidc client only takes the last redirect url from the configuration. Is doesn't match the base on the currently loaded host.

So if you want to use oidc for both apps (ex: admin.dsm.local and drive.dsm.local) have a different url, it won't works.

Ain't much a zitadel problem since the app is configured to accept both url as redirect.

**Reply by Aurel004** [reply]
I have the same issue unfortunately
Potential bypass is to set drive/photos etc to path and not subdomain: dsm.local/drive dsm.local/photos (set Alias and not Domain in the settings)

**Comment by pnsbttl** [comment]
![image](https://github.com/user-attachments/assets/40f230b3-bf33-42ae-b8ee-2a604dfff049)

I try to configure and I have this error...
created sub domain, as descibed but I don't know what I've missed...
Thanks you your help!

**Reply by Aurel004** [reply]
> ![image](https://github.com/user-attachments/assets/40f230b3-bf33-42ae-b8ee-2a604dfff049)
> 
> 
> 
> I try to configure and I have this error...
> 
> created sub domain, as descibed but I don't know what I've missed...
> 
> Thanks you your help!

Hi, it might be a ssl error if you're using a self signed certificate. I did not have this issue as I'm using a letsencrypt cert
Check if the syno allows the self signed cert of the zitadel subdomain

**Reply by pnsbttl** [reply]
Hi Aurel004,
it's a certificate connected with my domain, no self signed certificate.
From my understanding, i have to create record in reverse proxy, but to where this record should point?
Do I have to install Zitadel on Syno?

**Reply by Aurel004** [reply]
The record should point to Zitadel as the Synology will try to access it via the URL
Check if the Syno can resolve the Zitadel's full domain (or the well-known URL) using curl
Where is installed Zitadel on your setup ? Is it a Cloud Zitadel or self hosted ?

**Reply by pnsbttl** [reply]
Cloud Zitadel.
I've followed this configuration page: https://github.com/zitadel/zitadel/discussions/7907

**Reply by Aurel004** [reply]
Are you able to access your Zitadel instance from your Synology ?
